### ISSUE TITLE
Missing dependency for Axios 0.16.3

### ISSUE THREAD

**Issue by dustinaleksiuk** [contentful/contentful-management.js#141]
Our app has its contentful-management version set to 3.9.2 in an npm-shrinkwrap file. We realized today that this version depends on a version of Axios that seems to have been forked by Contentful but also that doesn't currently exist (0.16.3).

I went and looked at your fork and I can't see anything about 0.16.3 but I did find one other project on github with 0.16.3 referenced in their yarn.lock file and it's referencing your fork.

Any insight would be helpful. I'm nervous to npm install and regenerate my shrinkwrap since we have a lot of dependencies.

Edit: I got this working by creating a temporary project and generating a shrinkwrap.json file from a package.json with contentful-management 3.9.2. I then hacked the axios dependency from it into my existing shrinkwrap file.

My old dependency for Axios looked like this in the npm-shrinkwrap.json:

```
"axios": {
          "version": "0.16.3",
          "from": "contentful/axios#fix/https-via-http-proxy",
          "resolved": "git://github.com/contentful/axios.git#486c956313b4937d83202ae72304d3eb210487bb"
}
```
It now looks like this:

```
"axios": {
          "version": "0.16.2",
          "from": "contentful/axios#fix/https-via-http-proxy",
          "resolved": "git://github.com/contentful/axios.git#e9a9c62ce02115f7a4d7f197ba43362dcf0ea3d0"
 }
```

It makes me really nervous that I'm referencing specific commits that can disappear.

**Comment by Khaledgarbaya** [comment]
Hi @dustinaleksiuk,

We are trying our hard to get this PR merged with axios. see this [PR]. the commit is in a contentful fork so don't worry about it it will not be removed. Also, we distribute contentful-management with the dis folder and axios already backed in there.

We might consider changing Axios if this PR is not merged soon.

Best,
Khaled

**Comment by dustinaleksiuk** [comment]
Hi @Khaledgarbaya, thanks for that reply.

If you can, do you have an explanation of why commit [486c956313b4937d83202ae72304d3eb210487bb](https://github.com/axios/axios/commit/486c956313b4937d83202ae72304d3eb210487bb) used to work but now doesn't? As you can see in my post above, that commit used to be referenced through your fork but it now fails when I run npm install with the version locked to that commit. I'd just like to understand it to cover my risk since if we had tried to deploy to production on a new server with that lock file, it would have failed. It only works because that dependency is cached on older machines.

**Comment by Khaledgarbaya** [comment]
Hi @dustinaleksiuk I think the main problem is the `package-lock.json`.
We had a similar situation when publishing contentful-management npm package. once we deleted the lock files the install was correct an it was pulling the correct commit.

**Comment by dustinaleksiuk** [comment]
@Khaledgarbaya, I'm sort of with you on that. Since your package.json references a git branch, npm install does a revlist from git and grabs the most recent commit. The lock file then grabs that exact commit point for its own install to make sure that later installs use the same commit point. I'm not an expert at this at all. I spent my afternoon yesterday figuring it out.

So I'd argue that the problem is that a referenced commit that worked in July or so disappeared for some reason. It seems reasonable that the lock file should be able to lock in like that to ensure a consistent installation.

The reason I didn't want to re-run the install is because we're in production and I didn't have time to vet all of the versions that it'll try to update across all our packages.

Anyway, thanks very much for taking the time to respond.

**Comment by dustinaleksiuk** [comment]
I'd like to reopen this since it is still a problem for me. We use NPM 3.10.10 and we generate a shrinkwrap file. The Contentful axios fork is represented as a specific github commit in my npm-shrinkwrap.json file. 

That commit occasionally disappears. This one worked about 2 or 3 weeks ago:
a41f3ca97188a3a845d386bb140c424842379a91

I can't find that commit in your commit list in the /contentful/axios repo now.

I just regenerated the shrinkwrap and the new commit is this: 4b06f4a63db3ac16c99f7c61b584ef0e6d11f1af

I can find that one. It's here: https://github.com/contentful/axios/search?utf8=%E2%9C%93&q=4b06f4a63db3ac16c99f7c61b584ef0e6d11f1af&type=

It's pretty weird. My build wants those commits to be there and they seem to go away after a while. 

I'm hoping you guys are deleting commits somehow and you can simply stop doing that. Any insight would be really helpful to me.

**Comment by Khaledgarbaya** [comment]
Hi @dustinaleksiuk,
That should be fixed permanently now in `v4.1.2`.
Sorry about any inconveniences.

Best,
Khaled

**Comment by dustinaleksiuk** [comment]
Hi @Khaledgarbaya,

Thank you for that reply. I actually tried upgrading last week and I saw that my shrinkwrap file still references a git commit directly. I'll commit the upgrade though and hope that the axios git commits don't disappear.
